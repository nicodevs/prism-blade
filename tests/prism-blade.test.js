import { describe, expect, it } from 'vitest';
import Prism from 'prismjs';
import '../src/index.mjs';

// Helper to highlight Blade code
function highlightBlade(code) {
  return Prism.highlight(code, Prism.languages.blade, 'blade');
}

describe('Blade syntax highlighting', () => {
  it('should highlight comments', () => {
    const code = `{{-- resources/views/dashboard.blade.php --}}`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight directives', () => {
    const code = `@auth`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight directives with parameters', () => {
    const code = `@extends('layouts.app')`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight directives parameters as PHP', () => {
    const code = `@json(['user' => Auth::user(), 'notifications' => $notifications])`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight contents of the php directive as PHP', () => {
    const code = `@php
      $userRole = Auth::user()->role;
    @endphp`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight Blade interpolations', () => {
    const code = `<ul foo="bar">
        <li>Email: {{ Auth::user()->email }}</li>
        <li>Role: {{ $userRole }}</li>
    </ul>`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight Blade interpolations in attribute values', () => {
    const code = `<link href="{{ asset('css/dashboard.css') }}" rel="stylesheet">`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight unscaped Blade interpolations', () => {
    const code = `{!! Auth::user()->name !!}`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight components with attributes and shorthands', () => {
    const code = `<x-alert type="error" :name="auth()->user()->name" :message="$message->getMessage()" />`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should highlight scripts and style blocks', () => {
    const code = `@push('scripts')
        <style>
            .card { color: red; }
        </style>
        <script>
            document.addEventListener('DOMContentLoaded', function() {
                console.log('Loaded');
            });
        </script>
    @endpush`;
    expect(highlightBlade(code)).toMatchSnapshot();
  });

  it('should parse {{ and }} as token symbol', () => {
    const code = `{{ Auth::user()->email }}`;
    const highlighted = highlightBlade(code);
    expect(highlighted).toContain('<span class="token symbol symbol">{{</span>');
    expect(highlighted).toContain('<span class="token symbol symbol">}}</span>');
  });

  it('should parse {!! and !!} as token symbol', () => {
    const code = `{!! Auth::user()->name !!}`;
    const highlighted = highlightBlade(code);
    expect(highlighted).toContain('<span class="token symbol symbol">{!!</span>');
    expect(highlighted).toContain('<span class="token symbol symbol">!!}</span>');
  });

  it('should not highlight as Blade the contents of @verbatim', () => {
    const code = `<html>
        @verbatim
            Name of javascript {{ name }}
            Type of appliation {{ type }}
            <!-- and few more like these -->

            <script>console.log('Loaded');</script>
        @endverbatim
    </html>`;
    const highlighted = highlightBlade(code);
    expect(highlighted).toMatchSnapshot();
    expect(highlighted).toContain('Type of appliation {{ type }}');
    expect(highlighted).toContain('<span class="token comment">');
  });
});
