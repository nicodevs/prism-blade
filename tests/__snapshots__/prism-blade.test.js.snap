// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blade syntax highlighting > should highlight Blade interpolations 1`] = `
"<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>ul</span> <span class="token special-attr"><span class="token attr-name">foo</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string double-quoted-string">"bar"</span></span></span><span class="token punctuation">></span></span>
        <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>li</span><span class="token punctuation">></span></span>Email: <span class="token blade-echo-escaped variable"><span class="token symbol symbol">{{</span> <span class="token class-name static-context">Auth</span><span class="token operator">::</span><span class="token function">user</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">-></span><span class="token property">email</span> <span class="token symbol symbol">}}</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>li</span><span class="token punctuation">></span></span>
        <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>li</span><span class="token punctuation">></span></span>Role: <span class="token blade-echo-escaped variable"><span class="token symbol symbol">{{</span> <span class="token variable">$userRole</span> <span class="token symbol symbol">}}</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>li</span><span class="token punctuation">></span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>ul</span><span class="token punctuation">></span></span>"
`;

exports[`Blade syntax highlighting > should highlight Blade interpolations in attribute values 1`] = `"<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>link</span> <span class="token special-attr"><span class="token attr-name">href</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string double-quoted-string">"<span class="token blade-echo-escaped variable"><span class="token symbol symbol">{{</span> <span class="token function">asset</span><span class="token punctuation">(</span><span class="token string single-quoted-string">'css/dashboard.css'</span><span class="token punctuation">)</span> <span class="token symbol symbol">}}</span></span>"</span></span></span> <span class="token special-attr"><span class="token attr-name">rel</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string double-quoted-string">"stylesheet"</span></span></span><span class="token punctuation">></span></span>"`;

exports[`Blade syntax highlighting > should highlight comments 1`] = `"<span class="token comment">{{-- resources/views/dashboard.blade.php --}}</span>"`;

exports[`Blade syntax highlighting > should highlight components with attributes and shorthands 1`] = `"<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>x-alert</span> <span class="token special-attr"><span class="token attr-name">type</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string double-quoted-string">"error"</span></span></span> <span class="token special-attr"><span class="token attr-name">:name</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string"><span class="token punctuation">"</span><span class="token php-content"><span class="token function">auth</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">-></span><span class="token function">user</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">-></span><span class="token property">name</span></span><span class="token punctuation">"</span></span></span></span> <span class="token special-attr"><span class="token attr-name">:message</span><span class="token punctuation">=</span><span class="token attr-value"><span class="token string"><span class="token punctuation">"</span><span class="token php-content"><span class="token variable">$message</span><span class="token operator">-></span><span class="token function">getMessage</span><span class="token punctuation">(</span><span class="token punctuation">)</span></span><span class="token punctuation">"</span></span></span></span> <span class="token punctuation">/></span></span>"`;

exports[`Blade syntax highlighting > should highlight contents of the php directive as PHP 1`] = `
"<span class="token php-block"><span class="token php-start keyword">@php</span><span class="token php-content">
      <span class="token variable">$userRole</span> <span class="token operator">=</span> <span class="token class-name static-context">Auth</span><span class="token operator">::</span><span class="token function">user</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">-></span><span class="token property">role</span><span class="token punctuation">;</span>
    </span><span class="token php-end keyword">@endphp</span></span>"
`;

exports[`Blade syntax highlighting > should highlight directives 1`] = `"<span class="token blade-directive keyword">@auth</span>"`;

exports[`Blade syntax highlighting > should highlight directives parameters as PHP 1`] = `"<span class="token blade-directive-with-params"><span class="token directive-name keyword">@json</span><span class="token directive-params"><span class="token punctuation">(</span><span class="token punctuation">[</span><span class="token string single-quoted-string">'user'</span> <span class="token operator">=></span> <span class="token class-name static-context">Auth</span><span class="token operator">::</span><span class="token function">user</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token punctuation">,</span> <span class="token string single-quoted-string">'notifications'</span> <span class="token operator">=></span> <span class="token variable">$notifications</span><span class="token punctuation">]</span><span class="token punctuation">)</span></span></span>"`;

exports[`Blade syntax highlighting > should highlight directives with parameters 1`] = `"<span class="token blade-directive-with-params"><span class="token directive-name keyword">@extends</span><span class="token directive-params"><span class="token punctuation">(</span><span class="token string single-quoted-string">'layouts.app'</span><span class="token punctuation">)</span></span></span>"`;

exports[`Blade syntax highlighting > should highlight scripts and style blocks 1`] = `
"<span class="token blade-directive-with-params"><span class="token directive-name keyword">@push</span><span class="token directive-params"><span class="token punctuation">(</span><span class="token string single-quoted-string">'scripts'</span><span class="token punctuation">)</span></span></span>
        <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>style</span><span class="token punctuation">></span></span><span class="token style"><span class="token language-css">
            <span class="token selector">.card</span> <span class="token punctuation">{</span> <span class="token property">color</span><span class="token punctuation">:</span> red<span class="token punctuation">;</span> <span class="token punctuation">}</span>
        </span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>style</span><span class="token punctuation">></span></span>
        <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>script</span><span class="token punctuation">></span></span><span class="token script"><span class="token language-javascript">
            document<span class="token punctuation">.</span><span class="token function">addEventListener</span><span class="token punctuation">(</span><span class="token string">'DOMContentLoaded'</span><span class="token punctuation">,</span> <span class="token keyword">function</span><span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
                console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'Loaded'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>
            <span class="token punctuation">}</span><span class="token punctuation">)</span><span class="token punctuation">;</span>
        </span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>script</span><span class="token punctuation">></span></span>
    <span class="token blade-directive keyword">@endpush</span>"
`;

exports[`Blade syntax highlighting > should highlight unscaped Blade interpolations 1`] = `"<span class="token blade-echo-unescaped variable"><span class="token symbol symbol">{!!</span> <span class="token class-name static-context">Auth</span><span class="token operator">::</span><span class="token function">user</span><span class="token punctuation">(</span><span class="token punctuation">)</span><span class="token operator">-></span><span class="token property">name</span> <span class="token symbol symbol">!!}</span></span>"`;
